server/src/main/java/org/opensearch/rest/RestController.java

```diff
@@ -593,7 +593,7 @@ private boolean handleAuthenticateUser(final RestRequest request, final RestChan
                 // Authentication did not fail so return true. Authorization is handled at the action level.
                 return true;
             }
-            final Subject currentSubject = identityService.getSubject();
+            final UserSubject currentSubject = (UserSubject) identityService.getCurrentSubject();
```

Comment:
@reta I had to rely on casting here. This method is guarded behind the feature flag for identity.

IMO the `handleAuthenticateUser` should be removed from here and ActionPlugin.getRestHandlerWrapper should move to IdentityPlugin to keep the implementation consistent with how the security plugin works which is battle tested.

Reply:
Added a type check to this block